### ISSUE TITLE
Should st2timersengine default to local timezone?

### ISSUE THREAD

**Issue by nmaludy** [StackStorm/st2#4356]
##### SUMMARY

Curious why it's necessary to configure the timezone of the timers engine, instead of simply defaulting to the timezone of the local system?

Looking at the code, it doesn't seem to matter what the timezone is, because the timers are using the timezone from the rule definition. 

I'm asking for `puppet-st2`. Is this an essential config option, or is it OK to be defaulted to `America/Los_Angeles` everywhere?

##### ISSUE TYPE:
 - Feature Idea

##### STACKSTORM VERSION
`2.9dev`

**Comment by cognifloyd** [comment]
`ansible-st2` leaves the default `America/Los_Angeles` unless the playbook user explicitly changes it via the `st2_config.timer*.local_timezone` vars.

**Comment by armab** [comment]
cc @Kami which as I remember is in favor of UTC everywhere.

**Comment by Kami** [comment]
Yes, I believe all the services should always use UTC internally.

Local / custom timezone option should only be available for user facing stuff (CLI, WebUI, itd.). This way there is no confusion and potential date time conversion related issues when different objects are saved with different timezones, etc.

As far as this specific setting goes - it's there for backward compatibility reasons since we already supported this option before splitting timers engine into a separate service.

Having said that, I also think this value should default to UTC to avoid confusion.

**Comment by nmaludy** [comment]
Does the timezone even matter?

It looks like each rule specifies its own timezone. I'm confused, not sure what this setting actually affects?

**Comment by cognifloyd** [comment]
grepping the code, it looks like "local" timezone is a misnomer. It is the timezone used by the scheduler.
https://github.com/StackStorm/st2/blob/0245a18198babafe2ab0abde31791cf526435234/st2reactor/st2reactor/cmd/timersengine.py#L58-L60
https://github.com/StackStorm/st2/blob/f71f9b1ca03db1f9b916336f33be46a099900788/st2reactor/st2reactor/timer/base.py#L19
https://github.com/StackStorm/st2/blob/f71f9b1ca03db1f9b916336f33be46a099900788/st2reactor/st2reactor/timer/base.py#L40-L46
The APScheduler docs refer to this as:

**Comment by cognifloyd** [comment]
In conjunction with upgrading from 2.7.0 to 2.9.1, I just switched my production st2 instance from US/Central to UTC. I will report back if I notice anything weird. I suspect timers will work just as they did before - triggering at the appropriate timezone-specific time.

**Comment by nmaludy** [comment]
Still an issue

**Comment by nmaludy** [comment]
Still an issue.

**Comment by cognifloyd** [comment]
I've had my system in UTC for a long time without issue.
Next step: an internal st2 discussion to decide which way to take this:
- do nothing
- remove the config option and always use utc (my preferred option)
- remove the option and always use local system timezone
